src/ray/util/tests/shared_lru_test.cc

```diff
@@ -73,4 +85,13 @@ TEST(SharedLruConstCache, TypeAliasAssertion) {
       std::is_same_v<SharedLruConstCache<int, int>, SharedLruCache<int, const int>>);
 }
 
+TEST(SharedLruConstCache, CustomizedKey) {
+  TestClass obj1{"hello"};
+  TestClass obj2{"hello"};
+  SharedLruCache<TestClass, std::string> cache{2};
+  cache.Put(obj1, std::make_shared<std::string>("val"));
+  auto val = cache.Get(obj2);
```

Comment:
I guess we want to test cache.Get(absl::string_view{"hello"}) here?

Reply:
No, heterogeneous lookup is provided by flat hash map implementation.
To achieve that, we need to implement more features, like comparison between `TestClass` and `std::string`.
Reference: https://github.com/abseil/abseil-cpp/blob/c7cf999bda8390d2dd294ef903716a80135e6f4c/absl/container/flat_hash_map.h#L86-L94

Reply:
The intended use case for heterogeneous lookup here is C++ native string types, including std::string_view, std::string and const char*, all of them have been implemented by hash map.

Reply:
right, can we have a test case that heterogeneous lookup is used?

Reply:
I think I already did?
The above `PutAndGet` test case is using `std::string` for storage type, but `std::string_view` for delete and lookup type?